src/Elasticsearch/Metadata/Resource/Factory/ElasticsearchProviderResourceMetadataCollectionFactory.php

```diff
@@ -13,64 +13,46 @@
 
 namespace ApiPlatform\Elasticsearch\Metadata\Resource\Factory;
 
+use ApiPlatform\Elasticsearch\Metadata\Get;
+use ApiPlatform\Elasticsearch\Metadata\GetCollection;
+use ApiPlatform\Elasticsearch\Metadata\Operation as ElasticsearchOperation;
 use ApiPlatform\Elasticsearch\State\CollectionProvider;
 use ApiPlatform\Elasticsearch\State\ItemProvider;
 use ApiPlatform\Metadata\CollectionOperationInterface;
+use ApiPlatform\Metadata\HttpOperation;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
 use ApiPlatform\Metadata\Resource\ResourceMetadataCollection;
 use ApiPlatform\Util\Inflector;
 use Elasticsearch\Client;
-use Elasticsearch\Common\Exceptions\Missing404Exception;
 use Elasticsearch\Common\Exceptions\NoNodesAvailableException;
 
 final class ElasticsearchProviderResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
 {
-    public function __construct(private readonly Client $client, private readonly ResourceMetadataCollectionFactoryInterface $decorated)
-    {
+    public function __construct(
+        private readonly Client $client,
+        private readonly ResourceMetadataCollectionFactoryInterface $decorated,
+        private readonly array $mapping = [],
```

Comment:
This `mapping` argument isn't useful, we should deprecate the `mapping` usage in the configuration and recommend to use `defaults` instead.